gadget/install/install.go

```diff
@@ -176,36 +178,36 @@ func createFilesystem(part *gadget.OnDiskStructure, fsParams *mkfsParams, perfTi
 	return nil
 }
 
-func writePartitionContent(part *gadget.OnDiskStructure, fsDevice string, observer gadget.ContentObserver, perfTimings timings.Measurer) error {
-	partDisp := roleOrLabelOrName(part)
+func writePartitionContent(laidOut *gadget.LaidOutStructure, fsDevice string, observer gadget.ContentObserver, partDisp string, perfTimings timings.Measurer) error {
 	var err error
 	timings.Run(perfTimings, fmt.Sprintf("write-content[%s]", partDisp),
 		fmt.Sprintf("Write content for %s", partDisp),
 		func(timings.Measurer) {
-			err = writeFilesystemContent(part, fsDevice, observer)
+			err = writeFilesystemContent(laidOut, fsDevice, observer)
 		})
 	if err != nil {
 		return err
 	}
 	return nil
 }
 
-func installOnePartition(part *gadget.OnDiskStructure, encryptionType secboot.EncryptionType, sectorSize quantity.Size, observer gadget.ContentObserver, perfTimings timings.Measurer) (fsDevice string, encryptionKey keys.EncryptionKey, err error) {
+func installOnePartition(odls *onDiskAndLaidoutStructure, encryptionType secboot.EncryptionType, sectorSize quantity.Size, observer gadget.ContentObserver, perfTimings timings.Measurer) (fsDevice string, encryptionKey keys.EncryptionKey, err error) {
 	// 1. Encrypt
-	partDisp := roleOrLabelOrName(part)
-	fsParams, encryptionKey, err := maybeEncryptPartition(part, encryptionType, sectorSize, perfTimings)
+	part := odls.onDisk
+	partDisp := roleOrLabelOrName(odls.laidOut.Role, part)
```

Comment:
again  installOnePartition afaict is always called only with things that have a role, so maybe this can be simpler too

Reply:
That is true when called from FactoryReset(), but not when called from Run(), at least at first sight.

Reply:
afaiu `created` in Run contains things for which wasCreatedDuringInstall and IsCreatableAtInstall must be true, which means Role is not empty

Reply:
Right, I see also that `opts.CreateAllMissingPartitions` will be false in this case. I have changed that and also removed some checks in Run()/FactoryReset() that do not really apply in light of this.